### DISCUSSION TITLE
Event Bubbling

### DISCUSSION THREAD

**Discussion by danwize** [fabricjs/fabric.js#7114] [Q&A]
I'm rendering multi page pdfs with fabric.js.  All of the fabric canvases are contained in a div.  On mobile devices it is difficult to scroll through the pdf and can only be done if you drag on the area outside the canvases but still inside the divs.  Is it possible to have touch events bubble up out of fabric.js so the div can handle those and just scroll the whole div that way?

**Comment by melchiar** [comment]
Have you thought of adding a pan mode that toggles `pointer-events: none;` on the upper-canvas?

**Reply by danwize** [reply]
I tried pointer-events: none; on the upper-canvas class.  That that does prevent all the events from happening in fabric.js, but I still can't scroll.  It looks like that is stopping all the events.

**Comment by asturur** [comment]
if you can point out the part in the code that is stopping the events from bubbling out of fabricJS ( we probably do have a preventDefault somewhere ) we can make it optional.

**Reply by danwize** [reply]
I found several places where prevent default is called:

event.js line 66
canvas_events.mixin.js 216, 228, 289, 365
itext_click_behavior.mixin.js 50
itext_key_behavior.mixin.js 115, 145
itext_key_behaviour.js 245, 267, 280

I guess the itext code would not be of interested in this case, but perhaps changing it to be optional in event.js, and canvas_events.mixin.js?

Inside event.js in root.cancel I see root.stop and root.prevent are called.  Those prevent and stop methods are cancelling bubling and propogation and calling preventdefault.  I didn't see what calls root.cancel yet.
